src/main/java/com/medicalclinic/validator/PatientValidator.java

```diff
@@ -0,0 +1,21 @@
+package com.medicalclinic.validator;
+
+import static com.medicalclinic.exception.DictionaryHandler.getMessage;
+
+import com.medicalclinic.exception.ProcessingPatientException;
+import com.medicalclinic.model.Patient;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class PatientValidator {
+
+    public void validatePatientForUpdate(Patient patient, String referencedEmail) {
+        String idCardNo = patient.getIdCardNo();
+        if (idCardNo != null && !idCardNo.isBlank()) {
```

Comment:
Czy nie lepiej by było to rzucić w sytuacji jak po prostu jest różny niż aktualny?